payjoin/src/persist/mod.rs

```diff
@@ -0,0 +1,30 @@
+use serde::de::DeserializeOwned;
+use serde::Serialize;
+
+pub trait PersistableValue: Serialize + DeserializeOwned + Sized + Clone {}
+/// Implemented types that should be persisted by the application.
+pub trait Persister<V: PersistableValue> {
+    type Token;
```

Comment:
Can `Token` be something more generic? 
Token could be a trait object. Something the application can define or a conrete struct. 

```rust
type Token: Serialized + Deser
```

Reply:
Token: "wrapper type that enforces a typestate machine while allowing us to have a speration in calculating the affects and doing them".